app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -175,37 +180,47 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
     /** Adapter for the different timeline tabs */
     private lateinit var tabAdapter: MainPagerAdapter
 
+    @SuppressLint("RestrictedApi")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         val activeAccount = accountManager.activeAccount
             ?: return // will be redirected to LoginActivity by BaseActivity
 
         var showNotificationTab = false
-        if (intent != null) {
+
+        // check for savedInstanceState in order to not handle intent events more than once
+        if (intent != null && savedInstanceState == null) {
```

Comment:
All of the changes in MainActivity and other classes seem unrelated to this PR?

- Intent handling
- Context menu

Reply:
I think you're looking at the wrong diff. Those are changes from https://github.com/tuskyapp/Tusky/pull/3732 which has already been merged in to `develop` (and `develop` has been merged in to this PR).